portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -170,59 +170,48 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 
     var getCBioPortalMutationCounts = function (webservice_data) {
 	/* In: - webservice_data, a list of data obtained from the webservice API
-	 * Out: Promise which resolves with map from gene+","+start_pos+","+end_pos to cbioportal mutation count for that position range and gene
+	 * Out: Promise which resolves with map of type [gene]=>[codon position]=>mutation count
 	 */
-	var counts_map = {};
 	var def = new $.Deferred();
-	var to_query = {};
+	var to_query = {}; // [gene]=>[position]=>true, iff it should be queried
 	for (var i = 0; i < webservice_data.length; i++) {
 	    var datum = webservice_data[i];
 	    if (datum.genetic_alteration_type !== "MUTATION_EXTENDED" || datum.simplified_mutation_type !== "missense") {
 		continue;
 	    }
 	    var gene = datum.hugo_gene_symbol;
 	    var start_pos = datum.protein_start_position;
-	    var end_pos = datum.protein_end_position;
-	    if (gene && start_pos && end_pos && !isNaN(start_pos) && !isNaN(end_pos)) {
-		to_query[gene + ',' + parseInt(start_pos, 10) + ',' + parseInt(end_pos, 10)] = true;
+	    if (gene && start_pos && !isNaN(start_pos)) {
+		to_query[gene] = to_query[gene] || {};
+		to_query[gene][parseInt(start_pos, 10)] = true;
 	    }
 	}
-	var queries = Object.keys(to_query).map(function (x) {
-	    var splitx = x.split(',');
-	    return {
-		gene: splitx[0],
-		start_pos: splitx[1],
-		end_pos: splitx[2]
-	    };
-	});
-	var genes = queries.map(function (q) {
-	    return q.gene;
-	});
-	var starts = queries.map(function (q) {
-	    return q.start_pos;
-	});
-	var ends = queries.map(function (q) {
-	    return q.end_pos;
-	});
-
-	if (queries.length > 0) {
-	    window.cbioportal_client.getMutationCounts({
-		'type': 'count',
-		'per_study': false,
-		'gene': genes,
-		'start': starts,
-		'end': ends,
-		'echo': ['gene', 'start', 'end']
-	    }).then(function (counts) {
-		for (var i = 0; i < counts.length; i++) {
-		    var gene = counts[i].gene;
-		    var start = parseInt(counts[i].start, 10);
-		    var end = parseInt(counts[i].end, 10);
-		    counts_map[gene + ',' + start + ',' + end] = parseInt(counts[i].count, 10);
-		}
-		def.resolve(counts_map);
-	    }).fail(function () {
+	if (Object.keys(to_query).length > 0) {
+	    var query_genes = Object.keys(to_query);
+	    var query_positions = query_genes.map(function(gene) {
+		return Object.keys(to_query[gene]).map(function(pos) { return parseInt(pos, 10); });
+	    });
+	    $.ajax({
```

Comment:
The datamanager makes other ajax calls, for mutation annotation (oncokb, cosmic, ..), so I don't see this as out of scope of its responsibility.